validator_worker/src/sentry_interface.rs

```diff
@@ -378,3 +403,151 @@ pub mod campaigns {
+            .and(path(format!(
+                "/v5/channel/{}/spender/all",
+                DUMMY_CAMPAIGN.channel.id()
+            )))
+            .and(query_param("page", "0"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(&first_page_response))
+            .mount(&server)
+            .await;
+
+        Mock::given(method("GET"))
+            .and(path(format!(
+                "/v5/channel/{}/spender/all",
+                DUMMY_CAMPAIGN.channel.id()
+            )))
+            .and(query_param("page", "1"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(&second_page_response))
+            .mount(&server)
+            .await;
+
+        Mock::given(method("GET"))
+            .and(path(format!(
+                "/v5/channel/{}/spender/all",
+                DUMMY_CAMPAIGN.channel.id()
+            )))
+            .and(query_param("page", "2"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(&third_page_response))
+            .mount(&server)
+            .await;
+
+        let mut validators = Validators::new();
+        validators.insert(
+            DUMMY_VALIDATOR_LEADER.id,
+            Validator {
+                url: ApiUrl::from_str(&server.uri()).expect("Should parse"),
+                token: AuthToken::default(),
+            },
+        );
+        let mut config = configuration(Environment::Development, None).expect("Should get Config");
+        config.spendable_find_limit = 2;
+
+        let adapter = DummyAdapter::init(
+            DummyAdapterOptions {
+                dummy_identity: IDS["leader"],
+                dummy_auth: Default::default(),
+                dummy_auth_tokens: Default::default(),
+            },
+            &config,
+        );
+        let logger = discard_logger();
+
+        let sentry =
+            SentryApi::init(adapter, logger, config, validators).expect("Should build sentry");
+
+        let mut expected_response = HashMap::new();
+        expected_response.insert(ADDRESSES["user"], test_spender.clone());
+        expected_response.insert(ADDRESSES["publisher"], test_spender.clone());
+        expected_response.insert(ADDRESSES["publisher2"], test_spender.clone());
+        expected_response.insert(ADDRESSES["creator"], test_spender.clone());
+        expected_response.insert(ADDRESSES["tester"], test_spender.clone());
+
+        let res = sentry
+            .get_all_spenders(DUMMY_CAMPAIGN.channel.id())
+            .await
+            .expect("should get response");
+
+        assert!(expected_response.keys().len() == res.keys().len());
+        // Is page 1 included
+        assert!(res.contains_key(&ADDRESSES["user"]) && res.contains_key(&ADDRESSES["publisher"]));
+        // Is page 2 included
+        assert!(
+            res.contains_key(&ADDRESSES["publisher2"]) && res.contains_key(&ADDRESSES["creator"])
+        );
+        // Is page 3 included
+        assert!(res.contains_key(&ADDRESSES["tester"]));
```

Comment:
Assertions must be more that `assert!` contains key. If logic doesn't work => no additional info will be available on what has happened.

use something like `assert_eq!(page_one_expected, page_one_actual)`